From 55f8c95551890089db2560eae33efbf7d1b6ae42 Mon Sep 17 00:00:00 2001
From: Sean McBride <sean@rogue-research.com>
Date: Tue, 9 Apr 2024 13:42:14 -0400
Subject: [PATCH] descriptor: Fix addition overflow by correcting casts

The addition could overflow, the upcast needs to be performed before,
not after.

Fixes all clang-tidy bugprone-misplaced-widening-cast warnings

References #1479
---
 .clang-tidy           | 1 -
 libusb/descriptor.c   | 2 +-
 libusb/version_nano.h | 2 +-
 3 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/.clang-tidy b/.clang-tidy
index dc295d753..be101035f 100644
--- a/.clang-tidy
+++ b/.clang-tidy
@@ -3,7 +3,6 @@ Checks: "-*,\
 boost-*,\
 bugprone-*,\
 -bugprone-easily-swappable-parameters,\
--bugprone-misplaced-widening-cast,\
 -bugprone-narrowing-conversions,\
 -bugprone-signed-char-misuse,\
 -bugprone-switch-missing-default-case,\
diff --git a/libusb/descriptor.c b/libusb/descriptor.c
index 272ef6347..597868cd7 100644
--- a/libusb/descriptor.c
+++ b/libusb/descriptor.c
@@ -411,7 +411,7 @@ static int parse_configuration(struct libusb_context *ctx,
 		ptrdiff_t len = buffer - begin;
 		if (len > 0) {
 			uint8_t *extra = realloc((void *)config->extra,
-						 (size_t)(config->extra_length + len));
+						 (size_t)(config->extra_length) + (size_t)len);
 
 			if (!extra) {
 				r = LIBUSB_ERROR_NO_MEM;
diff --git a/libusb/version_nano.h b/libusb/version_nano.h
index a69807443..d8cd04b06 100644
--- a/libusb/version_nano.h
+++ b/libusb/version_nano.h
@@ -1 +1 @@
-#define LIBUSB_NANO 11923
+#define LIBUSB_NANO 11924
